.ci/test.sh

```diff
@@ -51,11 +51,20 @@ if [[ $TRAVIS == "true" ]] && [[ $TASK == "check-docs" ]]; then
 fi
 
 if [[ $TASK == "lint" ]]; then
-    conda install -q -y -n $CONDA_ENV pycodestyle pydocstyle
+    conda install -q -y -n $CONDA_ENV \
+        pycodestyle \
+        pydocstyle \
+        r-argparse \
+        r-lintr
     pip install --user cpplint
+    echo "linting Python code"
     pycodestyle --ignore=E501,W503 --exclude=./compute,./.nuget . || exit -1
     pydocstyle --convention=numpy --add-ignore=D105 --match-dir="^(?!^compute|test|example).*" --match="(?!^test_|setup).*\.py" . || exit -1
+    echo "linting C++ code"
     cpplint --filter=-build/include_subdir,-build/header_guard,-whitespace/line_length --recursive ./src ./include || exit 0
+    echo "linting R code"
+    Rscript ${BUILD_DIRECTORY}/.ci/lint_r_code.R \
+        --source-dir ${BUILD_DIRECTORY}
```

Comment:
Please add `|| exit -1` after the `Rscript` call

Reply:
We are guaranteed to get a nonzero exit code from the code I'm proposing, that is what this line of code does: https://github.com/jameslamb/LightGBM/blob/misc/r_code_style/.ci/lint_r_code.R#L64

`quit(status = 10)` (for example) causes an R script to `exit 10`

Do you want me to add the `|| exit -1` just as a source of documentation in `test.sh`?

Reply:
@jameslamb 
> Do you want me to add the || exit -1 just as a source of documentation in test.sh?

Yes, please do this to force Travis to fail when R script returns non-zero status. As the R script is called from the shell script, without that `|| exit -1` Travis will continue running the rest of the shell script.

Reply:
Ah I see. Sorry, not trying to be annoying about this.

Any reason we do not just set `set -e` at the top of `test.sh`? If we did that, any call exiting with a non-zero exit code would cause the script itself to exit with a non-zero and we wouldn't have to manually manage exit codes on all of these steps.

Reply:
If you are open to that I'd be happy to open a separate PR that adds a `set -e` and removes all the `|| exit -1` in this script.

Reply:
No problem!
> Any reason we do not just set set -e at the top of test.sh? 

IDK, let's try and see... Maybe we should allow something to fail.
But let's do not mix things: merge this PR with `|| exit -1` and then feel free to replace all `|| exit -1` with `set -e`.

Reply:
🤝 agreed!